arghandling.c

```diff
@@ -16,7 +16,9 @@
    by the number of arguments used.
    Too bad we can't just use getopt for all of this, but we don't have
    enough information to do it right.  */
-
+#ifdef __cplusplus
```

Comment:
Isn't this source file written in C, there is no need to use `exterm "C"` here (unless you are trying to C source files with a C++ compiler)

Reply:
yes, sometimes you may try to compile it with c++ compiler

Reply:
> yes, sometimes you may try to compile it with c++ compiler.

Understandible, but I don't think this workaround is good for code quality. Are there any other projects which use similar hacks?

> for example, you want to compile it in one line of command.

Maybe just consider taking a few seconds and use autotools to setup the build for you. We also have a few tests, autotools should make it easier to execute them.

Reply:
aha, magisk uses it. See https://github.com/topjohnwu/Magisk/pull/6759/files#diff-b70bfb027a8ef3ec4ab07a53a3b8ba83e63334ba3ce5ca6d9d773900ed452e5eR198

Reply:
autotool is good but I dont know if macOS allows us to do so.